### ISSUE TITLE
Scoped instances won't execute hooks

### ISSUE THREAD

**Issue by emilianomon** [elysiajs/elysia#158]
Hey!

I am facing this strange behavior where if I set an instance to `scoped`, those that are used by the instance are not triggered.

```typescript
const scoped = new Elysia({ scoped: true })
.use(app)
.use(authorizationMiddleware)
.get('/user', ({ auth }) => {

  console.log('User retrieved!'); // Triggers

  return {
    email: auth.user.email,
    id: auth.user.id,
  };
})
```

```typescript
export const app = new Elysia({ name: 'app' })
.onRequest(() => {
  console.log('"app" instance executed.'); // Won't be triggered.
})
```

```typescript
export const authorizationMiddleware = new Elysia()
.use(app)
.derive(async () => {
  
  // ...
  // This also won't be triggered.

  return {
    auth: {
      user,
    },
  };
});
```

Is this expected?

**Comment by SaltyAom** [comment]
This should be fixed in #103

**Comment by SaltyAom** [comment]
Should be fixed in the latest version.

Reproducible code:
```ts
import { Elysia } from '../src'

const app = new Elysia({ name: 'app' }).onRequest(() => {
	console.log('"app" instance executed.') // Executed
})

export const authorizationMiddleware = new Elysia()
	.use(app)
	.derive(async () => {
		return {
			auth: {
				user: ''
			}
		}
	})

const scoped = new Elysia({ scoped: true })
	.use(app)
	.use(authorizationMiddleware)
	.get('/user', ({ auth }) => {
		return auth
	})
	.listen(3000)
```

Feels free to reopen if the issue still persists.
